src/core/annotation.js

```diff
@@ -752,6 +754,26 @@ var TextWidgetAnnotation = (function TextWidgetAnnotationClosure() {
   return TextWidgetAnnotation;
 })();
 
+var ButtonWidgetAnnotation = (function ButtonWidgetAnnotationClosure() {
+  function ButtonWidgetAnnotation(params) {
+    WidgetAnnotation.call(this, params);
+
+    this.data.noToggleToOff = this.hasFieldFlag(
+      AnnotationFieldFlag.NOTOGGLETOOFF);
+    this.data.radio = this.hasFieldFlag(AnnotationFieldFlag.RADIO);
```

Comment:
Right. What do you suggest?

Note that I've already put a bit of logic in `annotation_layer.js` that does some kind of compliancy check:

```
if (!parameters.data.pushbutton) {
    if (parameters.data.radio) {
       return new RadioButtonWidgetAnnotationElement(parameters);
    } else {
       return new CheckboxWidgetAnnotationElement(parameters);
    }
}
```